include/fiction/algorithms/network_transformation/inverter_balancing.hpp

```diff
@@ -0,0 +1,320 @@
+//
+// Created by benjamin on 13.04.23.
+//
+
+#ifndef FICTION_INVERTER_BALANCING_HPP
+#define FICTION_INVERTER_BALANCING_HPP
+
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+#include "fiction/utils/name_utils.hpp"
+
+#include <mockturtle/traits.hpp>
+#include <mockturtle/utils/node_map.hpp>
+#include <mockturtle/views/fanout_view.hpp>
+#include <mockturtle/views/topo_view.hpp>
+
+#include <type_traits>
+
+#if (PROGRESS_BARS)
+#include <mockturtle/utils/progress_bar.hpp>
+#endif
+
+namespace fiction
+{
+
+namespace detail
+{
+
+template <typename Ntk>
+class inverter_balancing_impl
+{
+  public:
+    explicit inverter_balancing_impl(const Ntk& ntk_src) : ntk{ntk_src} {}
+
+    Ntk run()
```

Comment:
This function is quite extensive. Can we break it down into smaller ones that we put in the `detail` namespace or add as `private` member functions to this class?